Show the current note when rotating to portrait mode on large devices

hichamboushaba opened the pull request:
### Fix
Fixes #1239 
Currently when rotating a large device to portrait mode, and unless fullscreen was enabled, we will show just the list of notes, which is not ideal for continuing editing notes.
This PR fixes this by opening the editor when rotating to portrait mode if a note is selected.

### Test
Using a tablet and a Chrome device:
1. Open the app in landscape mode.
2. Select a note.
3. Rotate the device to portrait mode.
4. Confirm that the editor was opened using the last selected note.
5. Rotate to landscape mode.
6. Confirm that the dual-pane view is shown, and that note is still selected.

Repeat the previous steps using markdown notes, and confirm that the preview option is respected when rotating.

### Review
Only one developer is required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.

hichamboushaba committed:
Open the editor activity when rotating to portrait

hichamboushaba committed:
Pass markdown and preview options to NotesActivity when rotating

hichamboushaba committed:
Check if activity is still active before invalidating options menu

hichamboushaba commented on Simplenote/src/main/java/com/automattic/simplenote/NoteEditorFragment.java:
```diff
@@ -844,7 +844,9 @@ protected void showMarkdown() {
                 @Override
                 public void run() {
-                    requireActivity().invalidateOptionsMenu();
+                    if(getActivity() != null) {
+                        getActivity().invalidateOptionsMenu();
+                    }
```
When rotating the screen from portrait to landscape, then to portrait fast, this runnable will try to run after the fragment was detached.

theck13 replied:
```suggestion
                    if (!isDetached()) {
                        requireActivity().invalidateOptionsMenu();
                    }
```

If we're trying to avoid calling `invalidateOptionsMenu()` when the fragment is detached, would using `isDetached()` be better?

theck13 reviewed (commented):
The code works as described.  I noticed an odd side effect though.  When rotating the device on a large screen device, there was a weird effect that doesn't occur on `develop`.  I'm not sure how to describe it other than motion blur or a circular warping.  I'm guessing it's a consequence of the `onNoteSelected` statements in these changes.  The app is going from dual pane to single pane with the note list to single pane with the note editor all while rotating.  Let me know if you don't see what I'm talking about and I'll try to get a recording.

hichamboushaba committed:
Check if fragment is detached before invalidating options menu

Co-authored-by: Tyler Heck <theck13@users.noreply.github.com>

hichamboushaba commented:
> The code works as described. I noticed an odd side effect though. When rotating the device on a large screen device, there was a weird effect that doesn't occur on `develop`. I'm not sure how to describe it other than motion blur or a circular warping. I'm guessing it's a consequence of the `onNoteSelected` statements in these changes. The app is going from dual pane to single pane with the note list to single pane with the note editor all while rotating. Let me know if you don't see what I'm talking about and I'll try to get a recording.

I don't see anything on my side, but I'm using just the emulator, as I don't have a physical tablet, a recording will be helpful.
Does it occur when going from portrait to landscape, or the inverse? And can you see if the same thing is happening on develop when enabling fullscreen editing in landscape mode?

theck13 reviewed (commented):
I'm seeing it with an emulator too using a Nexus 10 API 29 stock emulator.  It appears to be happening when rotating from portrait to landscape and from landscape to portrait.  I took videos of the emulator with the debug (2.14-rc-1) and release (2.13.0) versions of the app.  The debug version includes these changes.  The release version does not.  The blur/warp effect I'm talking about first occurs 1-2 seconds in both videos.  If you pause and screenshot the video while it's occurring, it looks like the effect is caused by the app overlaying multiple screens.  That exists in the release version, but it appears to be only two screens.  It's more noticeable in the debug version since it appears to have one or two more screens than the release version.  In the debug screenshot, you there are three floating action buttons.  In the release screenshot, there is only one.  See the screenshots and videos below for illustration.

Debug|Release
-|-
<img width="1254" alt="1266_editor_rotate_large_debug_6s" src="https://user-images.githubusercontent.com/3827611/103811756-a4409600-501a-11eb-9391-5bef57be2369.png">|<img width="1254" alt="1266_editor_rotate_large_release_6s" src="https://user-images.githubusercontent.com/3827611/103811761-a6a2f000-501a-11eb-9922-dc03928859a8.png">
[Video(Debug)](https://user-images.githubusercontent.com/3827611/103800889-342a1400-500a-11eb-819f-11bb633df430.mov)|[Video(Release)](https://user-images.githubusercontent.com/3827611/103800897-35f3d780-500a-11eb-88f2-86b8bda19186.mov)

The effect itself doesn't bother me too much.  I'm worried about non-high-end or old devices that may not be able to handle this behavior.  I don't have an actual device above Android 6.0.0 that is old or isn't high-end to test.

hichamboushaba committed:
Disable Activity animations when switching between dual and single pane modes

The activity transitions causes a motion blur when rotating the device, and switching between sinle and dual pane modes.

hichamboushaba commented:
I was able to reproduce the effect using the Nexus 10 emulator, I didn't have it in the Pixel C one.
I made a change on the commit dd4ef68 to make sure activity transition animations are disabled, it improves the situation, I can still see the blur sometimes, but most of the time it's not there.

And just for information, the same effect can be reproduced in the release version, using one the following steps: "Open a note in portrait mode, then rotate to landscape."

> I'm worried about non-high-end or old devices that may not be able to handle this behavior.

As the effect is already present in the release version using the above steps, and no user complained about it, I don't think it would cause any harm, except dropping some frames if the device can't handle it.

theck13 commented on Simplenote/src/main/java/com/automattic/simplenote/NoteEditorActivity.java:
```diff
@@ -277,11 +277,13 @@ public void onConfigurationChanged(@NonNull Configuration newConfig) {
-            resultIntent.addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION);
             setResult(Activity.RESULT_OK, resultIntent);
             finish();
+            overridePendingTransition(0,0);
```
```suggestion
            overridePendingTransition(0, 0);
```

There is a space missing between the parameters.

theck13 commented on Simplenote/src/main/java/com/automattic/simplenote/NotesActivity.java:
```diff
@@ -1413,9 +1408,10 @@ public void onConfigurationChanged(@NonNull Configuration newConfig) {
                 invalidateOptionsMenu();
-            // Go to NoteEditorActivity if note editing was fullscreen and orientation was switched to portrait
-            } else if (mNoteListFragment.isHidden() && mCurrentNote != null) {
+                // Go to NoteEditorActivity if a note was selected and orientation was switched to portrait
```
```suggestion
            // Go to NoteEditorActivity if a note was selected and orientation was switched to portrait
```

It looks like autoformat added an extra indent to this comment.

hichamboushaba committed:
Add missing space

Co-authored-by: Tyler Heck <theck13@users.noreply.github.com>

hichamboushaba committed:
Fix indention of the comment

Co-authored-by: Tyler Heck <theck13@users.noreply.github.com>
